app/utils/Messages.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+
+import scala.xml.NodeSeq
+
+object Messages {
+
+  def generateIE928Message(): NodeSeq = {
```

Comment:
No issue with what has been done here, but have you spent any time to see if there are any libraries that will follow the rules of an XSD to create an xml document? It would save us a lot of code and maintenance if we can find one.

Reply:
Also, don't just stick this in the utils package. I'd recommend you work from assuming each message can have its own file, simply because this will be quite big if you aren't using an XSD-based generator.

Reply:
The issue is with the values. It may be possible to generate the structure but I can find a reliable way to generate the values based on the validation regexes found in the XSD files.<br><br>I don't think there's a lot of message types to support so I think using a single file is the best approach. The TestXml trait in the API also works like this.

Reply:
I mean, fair on the single file.

Hmm. That's frustrating. Still, might be worth noting as an improvement in the future. Put together a ticket and link it here and I'll consider it addressed.